### PR TITLE
webapps error handling improvements

### PR DESCRIPTION
**Pull request by snopoke**
## Technical Summary
I noticed that errors from formplayer are often shown as 'unexpected error' which is the default text we use when we can't find another error message.

This PR does a few things (see commits) but the main goal is to do a better job of showing error messages when they exist and standardize the default errors.

### Safety story
Small changes to error message handling.

### Automated test coverage
None

### QA Plan
None

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by snopoke** [comment]
@orangejenny I get this build error:
```
AssertionError: hqImport used in RequireJS modules: 
  cloudcare/js/formplayer/constants imported in /mnt/commcare-hq-ro/corehq/apps/cloudcare/static/cloudcare/js/util.js
```

But if I change the code to use RequireJS imports then the imported module is `undefined`. Any suggestions on how to resolve this?

**Comment by orangejenny** [comment]
(I lost track of this PR, will take a look tomorrow)

**Comment by orangejenny** [comment]
(had time today after all)

For the build error, you should be able to add `cloudcare/js/formplayer/constants` to the list of requirements at the very top of the file, and then referenced the named module. If you already did that and had a circular reference or something, you can use `hqRequire` in the same place you're now using `hqImport` (little bit of docs on that [here](https://commcare-hq.readthedocs.io/js-guide/migrating.html?highlight=hqrequire#troubleshooting-migration-issues)).

**Comment by snopoke** [comment]
@orangejenny thanks, it turned out to be an issue with the JS file ordering on the page. It's working now.

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/constants.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ hqDefine("cloudcare/js/formplayer/constants", function () {
         WEB_APPS_ENVIRONMENT: 'web-apps',
         PREVIEW_APP_ENVIRONMENT: 'preview-app',
         GENERIC_ERROR: gettext(
-            'Formplayer encountered an error. ' +
```
Comment (orangejenny) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/constants.js:9]: 👍

**Review by orangejenny — APPROVED** [review]

**Pull request merged**